docs/getting_started/tutorial.rst

```diff
@@ -606,6 +609,237 @@ something like this:
+
+    class BlogCategoryAdmin(ModelAdmin):
+        model = BlogCategory
+        add_to_settings_menu = True
+        list_display = ('name', 'slug')
+
+    modeladmin_register(BlogCategoryAdmin)
+
+``wagtail_hooks`` lets you control aspects of the admin, and to expose non-Wagtail models.
+In this example, we've specified:
+
+``add_to_settings_menu = True``
+
+So that our BlogCategories appear in the global Settings menu:
+
+.. figure:: ../_static/images/tutorial/tutorial_11.png
+   :alt: Adding Blog Categories to Settings
+
+.. figure:: ../_static/images/tutorial/tutorial_12.png
+  :alt: Categories listing
+
+After using your new Blog Categories interface to create some categories, you can select them from the
+InlinePanel in a BlogPage:
+
+.. figure:: ../_static/images/tutorial/tutorial_13.png
+  :alt: Newly created categories available to a BlogPage
+
+Now that we're storing categories on posts, we need a view to display them, and a way to link to them.
+Rather than create another model for the new view, let's consider a category to be a "slice" of data exposed
+on the ``BlogIndexPage.``  We can pass a category to the view either as URL parameter: ``/blog?cat=science``
+or as a keyword on the end of the URL, which is much cleaner: ``/blog/science``. To access that keyword, we'll
+take advantage of Wagtail's :doc:`RoutablePageMixin <../reference/contrib/routablepage>`  class. Modify
+``BlogIndexPage`` like this:
+
+.. code-block:: python
+
+    from wagtail.contrib.wagtailroutablepage.models import RoutablePageMixin, route
+    from django.shortcuts import get_object_or_404, render
+
+    class BlogIndexPage(RoutablePageMixin, Page):
+        intro = RichTextField(blank=True)
+
+        def get_context(self, request):
+            # Update context to include only published posts, ordered by reverse-chron
+            context = super(BlogIndexPage, self).get_context(request)
+            blogpages = self.get_children().live().order_by('-first_published_at')
+
+            # Include queryset of non-empty blog categories for menu
+            usedcats = BlogCategoryBlogPage.objects.distinct().values_list('category__slug', flat=True)
+            blogcats = BlogCategory.objects.filter(slug__in=usedcats)
+
+            context['blogpages'] = blogpages
+            context['blogcats'] = blogcats
+            return context
+
+        @route(r'^cat/(\w+)/$', name="blog_category")
+        def category(self, request, catslug=None):
+            """
+            Filter BlogPages by category
+            """
+            category = get_object_or_404(BlogCategory, slug=catslug)
+            blogpages = BlogPage.objects.filter(
+                blog_categories__category=category).live().order_by('-first_published_at')
+
+            context = self.get_context(request)
+            context['blogpages'] = blogpages
+            context['category'] = category
+            return render(request, 'blog/blog_index_page.html', context)
```

Comment:
I was unsure about this part - is this the best way to render from a RoutablePageMixin method? Do we really have to specify the template name? Wagtail docs for RoutablePageMixin leave out sample code, so it's ambiguous.